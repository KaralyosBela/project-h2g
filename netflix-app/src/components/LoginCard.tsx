import React, { useState } from "react";
import classes from "./LoginCard.module.css";
import { useNavigate } from "react-router-dom";

export const LoginCard: React.FC = () => {
  const navigate = useNavigate();
  const [username, setUsername] = useState<string>("");
  const [password, setPassword] = useState<string>("");

  const usernameOnChange = (event: React.ChangeEvent<HTMLInputElement>) => setUsername(event.currentTarget.value);
  const passwordOnChange = (event: React.ChangeEvent<HTMLInputElement>) => setPassword(event.currentTarget.value);

  const auth = (event: any) => {
    event.preventDefault();
    if (username === "admin" && password === "admin") {
      navigate("/");
    }
  };

  return (
    <div className={classes.logincard}>
      <div className={classes.card}>
        <h1>LOG IN</h1>
        <form className={classes.inputs} onSubmit={auth}>
          <label htmlFor="username">USER ID*</label>
          <input type="text" id="username" onChange={usernameOnChange}></input>

          <label htmlFor="password">PASSWORD*</label>
          <input
            type="password"
            id="password"
            onChange={passwordOnChange}
          ></input>

          <div className={classes.action}>
            <button className={classes.resetBtn}>RESET</button>
            <button className={classes.loginBtn} type="submit">
              LOG IN
            </button>
          </div>

          <p>*Mandatory</p>
        </form>
      </div>
    </div>
  );
};
