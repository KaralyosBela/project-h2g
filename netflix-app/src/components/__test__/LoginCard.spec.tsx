import { fireEvent, render, screen } from "@testing-library/react";
import { LoginCard } from "../LoginCard";

const mockDispatch = jest.fn();
const mockUseDispatch = jest.fn();
const mockNavigate = jest.fn();
const mockUseNavigate = jest.fn();
jest
  .mock("react-redux", () => ({ useDispatch: () => mockDispatch }))
  .mock("react-router-dom", () => ({ useNavigate: () => mockNavigate }));
mockDispatch.mockReturnValue(mockUseDispatch);
mockNavigate.mockReturnValue(mockUseNavigate);

describe("Login component", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should render the login compoenent", () => {
    const { container } = render(<LoginCard />);
    expect(container).toMatchSnapshot();
  });

  it("should call navigate when authenticated", () => {
    render(<LoginCard />);

    const userInput = screen.getByTestId("userIdInput");
    const passwordInput = screen.getByTestId("passwordInput");
    const form = screen.getByRole("form");
    fireEvent.change(userInput, { target: { value: "admin" } });
    fireEvent.change(passwordInput, { target: { value: "admin" } });
    fireEvent.submit(form);

    expect(mockNavigate).toBeCalledTimes(1);
    expect(mockNavigate).toBeCalledWith("/");
  });

  it("should not call navigate when not authenticated", () => {
    render(<LoginCard />);

    const userInput = screen.getByTestId("userIdInput");
    const passwordInput = screen.getByTestId("passwordInput");
    const form = screen.getByRole("form");
    fireEvent.change(userInput, { target: { value: "asd" } });
    fireEvent.change(passwordInput, { target: { value: "asd" } });
    fireEvent.click(form);

    expect(mockNavigate).toBeCalledTimes(0); //or expect(mockNavigate).not.toBeCalled();
  });
});
