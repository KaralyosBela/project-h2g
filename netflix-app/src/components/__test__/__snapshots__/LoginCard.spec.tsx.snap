// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should render the login compoenent 1`] = `
<div>
  <div
    class="logincard"
  >
    <div
      class="card"
    >
      <h1>
        LOG IN
      </h1>
      <form
        class="inputs"
        role="form"
      >
        <label
          for="username"
        >
          USER ID*
        </label>
        <input
          data-testid="userIdInput"
          id="username"
          type="text"
        />
        <label
          for="password"
        >
          PASSWORD*
        </label>
        <input
          data-testid="passwordInput"
          id="password"
          type="password"
        />
        <div
          class="action"
        >
          <button
            class="resetBtn"
          >
            RESET
          </button>
          <button
            class="loginBtn"
            data-testid="loginBtn"
            type="submit"
          >
            LOG IN
          </button>
        </div>
        <p>
          *Mandatory
        </p>
      </form>
    </div>
  </div>
</div>
`;
